marker_error/src/trace.rs

```diff
@@ -0,0 +1,68 @@
+use std::backtrace::{Backtrace, BacktraceStatus};
+use std::fmt;
+use tracing_error::SpanTrace;
+use yansi::Paint;
+
+/// Contains a verbose information useful for debugging. The developers
+/// will most likely want to see this when a panic happens in their code or
+/// an error is returned from a function where it shouldn't be returned.
+#[derive(Debug)]
+pub struct ErrorTrace {
+    spantrace: SpanTrace,
+    backtrace: Backtrace,
+}
+
+impl ErrorTrace {
+    pub fn capture() -> Self {
+        Self {
+            spantrace: SpanTrace::capture(),
+            backtrace: Backtrace::force_capture(),
+        }
+    }
+
+    fn write_trace(f: &mut fmt::Formatter<'_>, kind: &str, trace: &dyn fmt::Display) -> fmt::Result {
+        writeln!(f, "{}\n{trace}", format_args!("{kind}:").red().bold())
+    }
+}
+
+impl fmt::Display for ErrorTrace {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let backtrace = &self.backtrace;
+
+        match backtrace.status() {
+            BacktraceStatus::Captured => Self::write_trace(f, "Backtrace", backtrace)?,
+            // This must never happen because we use `force_capture`, but let's
+            // be safe and not panic the second time in a row here if we this ever breaks
+            BacktraceStatus::Disabled => writeln!(
+                f,
+                "{}",
+                "⚠️  Backtrace wasn't captured. t is probably a bug in our error \
+                trace capturing code."
+                    .red(),
+            )?,
+            BacktraceStatus::Unsupported => writeln!(f, "{}", "backtrace is unsupported".red())?,
+            _ => writeln!(f, "{}\n{backtrace}", "Backtrace is in unknown status:".red())?,
+        };
+
+        let spantrace = &self.spantrace;
+
+        match spantrace.status() {
+            tracing_error::SpanTraceStatus::CAPTURED => {
+                write!(f, "{}\n{spantrace}", "Spantrace:".red().bold())?;
+            },
+            tracing_error::SpanTraceStatus::EMPTY => {
+                write!(
+                    f,
+                    "{}. It may be expected if the error happened outside of \
+                    any spans, or those spans were not enabled. You may try \
+                    capturing it by enabling {} logging or increasing its level.",
+                    "⚠️  Spantrace wasn't captured".yellow(),
+                    "tracing".bold()
+                )?;
```

Comment:
There is a dilemma here. This crate is used in several different binaries. Every such binary has its own env variable used for enabling the tracing. For example, `cargo-marker` uses `MARKER_LOG`, but `marker_rustc_driver` uses `RUSTC_LOG`, because it uses `rustc_log` crate to initialize `tracing`.
So if I were to hardcode `MARKER_LOG` in the error message here, then it would be wrong in the context of `RUSTC_LOG`. I considered using a global variable to deliver the info about the tracing env var used to this crate, but meh. 
The other compromise option is to write "Enable spantrace with `MARKER_LOG` for `cargo_marker` or `RUSTC_LOG` for `marker_rustc_driver`".

But... There is another caveat. Spantraces will never be captured for `marker_rustc_driver`. I haven't figured out how to do that yet. The problem is that `marker_rustc_driver` uses `rustc_log` crate that uses some custom internal `tracing` subscribers which I wouldn't like to override, but I also can't extend the `tracing` setup because `rustc_log` uses `tracing::subscriber::set_global_default()` [here](https://github.com/rust-lang/rust/blob/f6faef447536204549eb47f12884d3e229fe87d3/compiler/rustc_log/src/lib.rs#L95-L99). This isn't composable. once the tracing subscriber is it's possible to get it back, but it's not possible to extend it.

So even if I would say "Use `RUSTC_LOG` to enable spantrace capturing", then this wouldn't be true, because there isn't an `ErrorLayer` in the tracing subscriber for `marker_rustc_driver` and it won't work. 

It may be possible to modify `rustc_log` crate to expose the API that enables adding an `ErrorLayer` to the subscriber, but I didn't do it in this scope. I don't think it is critical, and may be done any time layer.

For now I can add "Use `MARKER_LOG` for cargo-marker; spantrace capturing in `marker_rustc_driver` is not supported yet"